store/event_test.go

```diff
@@ -84,6 +88,15 @@ func TestScanHistory(t *testing.T) {
 	if e != nil {
 		t.Fatalf("bad index shoud reuturn nil")
 	}
+
+	// Add a dir which will replace old event
+	eh.addEvent(newEvent(Create, "/foo/bar/bar/bar", 7, 7))
+
+	// test for the event which has been replaced.
+	_, err = eh.scan("/foo", false, 1)
+	if err == nil || err.ErrorCode != etcdErr.EcodeEventIndexCleared {
```

Comment:
@gyuho  you are right the nil check is not required. Will remove it.
@xiang90 Please review.